### PR TITLE
test: update tests for checking if rpm is signed when we use -PGPG_KEY

### PR DESCRIPTION
**Pull request by zdtsw**
- GHA test case without gpg
- https://ci.adoptopenjdk.net/job/Sophia-adoptium-packages-linux-pipeline/144/artifact/jdk/redhat/build/reports/tests/checkJdkRedHat/classes/packaging.DnfOperationsTest.html test case with -PGPG_KEY set : to have a negative debug to print assert `Signature   : RSA/SHA256, Wed Aug  3 15:13:10 2022, Key ID 843c48a565f8f04b` should be matched
- https://ci.adoptopenjdk.net/job/Sophia-adoptium-packages-linux-pipeline/143/ test cas with -PGPG_KEY set: to have a positive case
- uplift dependency version : most were released 1.5yrs ago to latest ones

Ref: https://github.com/adoptium/installer/issues/513

### CODE REVIEW

File: linux/jdk/redhat/build.gradle [hunk 1]
```diff
@@ -108,8 +108,15 @@ task checkJdkRedHat(type: Test) {
...
 	def product = getProduct()
 	def productVersion = getProductVersion()
+	def gpgKey = getGPGKey()
+
 
+	
 	environment "PACKAGE", "$product-$productVersion-jdk"
+	if (gpgKey != null) {
+		environment "JDKGPG", "$gpgKey"
+		println "We set gpg!"
```
Comment (karianna) [inline linux/jdk/redhat/build.gradle:118]:
```suggestion
		println "We set the gpgkey!"
```

File: linux/jdk/suse/build.gradle [hunk 2]
```diff
@@ -112,8 +112,11 @@ task checkJdkSuse(type: Test) {
 	def gpgKey = getGPGKey()
 
 	environment "PACKAGE", "$product-$productVersion-jdk"
-	environment "GPG", "$gpgKey"
-
+	if (gpgKey != null) {
+		environment "JDKGPG", "$gpgKey"
+		println "We set gpg!"
```
Comment (karianna) [inline linux/jdk/suse/build.gradle:117]:
```suggestion
		println "We set the gpgkey!"
```

**Review by karianna — CHANGES_REQUESTED** [review]
small type updates

**Comment by zdtsw** [comment]
@smlambert can I get a review on this PR?

**Review by karianna — APPROVED** [review]

**Pull request merged**